tests/test_advection.py

```diff
@@ -134,12 +134,13 @@ def SubmergeParticle(particle, fieldset, time):
         particle.depth = 0
         AdvectionRK4(particle, fieldset, time)  # perform a 2D advection because vertical flow will always push up in this case
         particle.time = time + particle.dt  # to not trigger kernels again, otherwise infinite loop
+        particle.succeeded()
```

Comment:
This `self.succeeded` is only used in this unit tests? Why not directly assign ?
```suggestion
        particle.status = ErrorCode.Success
```

Reply:
well, ideally setting the status is done via function call and NOT via direct setting. Has to to with that variable being marked as 'read_only' in particle.py. What's your take on that ? I think it's actually a good idea to only modify the status internally and handle it just like a private member variable. Also, that (i.e. modifying the state with some code) is really not something the user shall be doing ...